projects/plugins/jetpack/extensions/blocks/ai-assistant/supports/index.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+import { addFilter } from '@wordpress/hooks';
+/**
+ * Internal dependencies
+ */
+import { isPossibleToExtendBlock } from '../extensions/ai-assistant';
+import withAIAssistant from '../extensions/ai-assistant/with-ai-assistant';
+
+export const SUPPORT_NAME = 'jetpack/ai';
+
+function handleJetpackAISupports( settings ) {
+	const jetpackAISupports = getBlockSupport( settings, SUPPORT_NAME );
+	if ( ! jetpackAISupports ) {
+		return settings;
+	}
+
+	// Check specific for the `assistant` support.
+	if ( ! jetpackAISupports?.assistant ) {
+		return settings;
+	}
+
+	if ( ! isPossibleToExtendBlock() ) {
```

Comment:
Any reason for calling it two times, here and on https://github.com/Automattic/jetpack/pull/30908/files#diff-75e6a34584392dbce5186fb1b565b3bc098000fe9c8f44afc583daf3a682dd40R67?